### ISSUE TITLE
add RandomApply and RandomChoice

### ISSUE THREAD

**Issue by innat** [keras-team/keras#20727]
This was included in keras-cv. 

- https://github.com/keras-team/keras-cv/pull/434
- https://github.com/keras-team/keras-cv/pull/1979 [random-apply](https://github.com/keras-team/keras-cv/blob/94b0a551d03ee9de3d81663e2c1a680cb113f7f1/keras_cv/src/layers/preprocessing/random_apply.py#L22)

**Comment by mehtamansi29** [comment]
Hi @innat -

Can you elaborate the issue which you are facing with more details ?

**Comment by innat** [comment]
I believe the two layers mentioned above haven't been moved from keras-cv to keras, but they should be.

**Comment by hertschuh** [comment]
> I believe the two layers mentioned above haven't been moved from keras-cv to keras, but they should be.

Hi @innat , correct they should be moved to keras. This work still needs to be done.

**Comment by harshaljanjani** [comment]
Hi!
Would love to contribute here if this task is up for the taking. I noticed in `#434` a comment mentioning that `RandomChoice` is currently under the `keras_cv.layers.preprocessing` module, and it seems `RandomApply` is there as well. While they’re not strictly preprocessing layers, when moving them, would they still go under `layers.preprocessing`?

**Comment by harshaljanjani** [comment]
Hey!
Just wanted to mention that this PR might require an API modification, so a modification of the labels in the issue thread would be much appreciated. Thanks!

**Comment by innat** [comment]
@fchollet 
This [PR](https://github.com/keras-team/keras/pull/21082) was closed due to enough reason to have these layers (random-choice, and random-apply). Below are the initial discussion regarding these operations. Could you please take a quick look at them and give your feedback? 

- random-choice: https://github.com/keras-team/keras-cv/issues/408
- random-apply (initially named maybe-apply): https://github.com/keras-team/keras-cv/pull/435

I initially requested to add `random-choice` so I could provide more details about its usefulness.
